libnetwork/internal/rootlessnetns/netns_linux.go

```diff
@@ -143,6 +155,38 @@ func (n *Netns) cleanup() error {
 	return multiErr.ErrorOrNil()
 }
 
+func (n *Netns) setupPasta(nsPath string) error {
+	pidPath := n.getPath(rootlessNetNsConnPidFile)
+
+	pastaOpts := pasta.SetupOptions{
+		Config:       n.config,
+		Netns:        nsPath,
+		ExtraOptions: []string{"-p", pidPath},
+	}
+	if err := pasta.Setup(&pastaOpts); err != nil {
+		return err
+	}
+
+	if systemd.RunsOnSystemd() {
+		// Treat these as fatal - if pasta failed to write a PID file something is probably wrong.
+		pidfile, err := os.ReadFile(pidPath)
+		if err != nil {
+			return fmt.Errorf("unable to open pasta PID file: %w", err)
+		}
+		pid, err := strconv.Atoi(string(pidfile))
+		if err != nil {
+			return fmt.Errorf("unable to decode pasta PID (got %q): %w", string(pidfile), err)
```

Comment:
no need to include the pidfile data in the error as the Atoi error already includes this so you would stutter the error.